### PR TITLE
MS-1082

### PR DESCRIPTION
**Pull request by azagan-meyvn**
Added "/byHcPartyPatientForeignKeys" post route in ContactFacade in order to retrieve all contacts related to a list of patients. 

Add filtering on health elements by tags and/or by status

### CODE REVIEW

File: src/main/java/org/taktik/icure/services/external/rest/v1/facade/ContactFacade.java [hunk 1]
```diff
@@ -270,6 +270,32 @@ public Response findByHCPartyFormIds(@QueryParam("hcPartyId") String hcPartyId,
...
+    public Response findByHCPartyPatientForeignKeys(@QueryParam("hcPartyId") String hcPartyId, ListOfIdsDto patientForeignKeys) {
+        if (hcPartyId == null || patientForeignKeys == null || patientForeignKeys.getIds().size()==0) {
+            return Response.status(400).type("text/plain").entity("A required query parameter was not specified for this request.").build();
+        }
+
+        List<Contact> contactList = contactLogic.findByHCPartyPatient(hcPartyId, patientForeignKeys.getIds());
+
+        boolean succeed = (contactList != null);
+        if (succeed) {
+
```
Comment (bsafta-meyvn) [inline src/main/java/org/taktik/icure/services/external/rest/v1/facade/ContactFacade.java]: Remove this extra line

File: src/main/java/org/taktik/icure/services/external/rest/v1/facade/ContactFacade.java [hunk 2]
```diff
@@ -270,6 +270,32 @@ public Response findByHCPartyFormIds(@QueryParam("hcPartyId") String hcPartyId,
...
+            // mapping to Dto
```
Comment (bsafta-meyvn) [inline src/main/java/org/taktik/icure/services/external/rest/v1/facade/ContactFacade.java]: Comment is not necessary, code is self explanatory

File: src/main/java/org/taktik/icure/services/external/rest/v1/facade/ContactFacade.java [hunk 3]
```diff
@@ -270,6 +270,30 @@ public Response findByHCPartyFormIds(@QueryParam("hcPartyId") String hcPartyId,
...
+    @ApiOperation(
+            value = "List contacts found By Healthcare Party and Patient foreign keys.",
+            response = ContactDto.class,
+            responseContainer = "Array",
+            httpMethod = "POST"
+    )
+    @POST
+    @Path("/byHcPartyPatientForeignKeys")
+    public Response findByHCPartyPatientForeignKeys(@QueryParam("hcPartyId") String hcPartyId, ListOfIdsDto patientForeignKeys) {
+        if (hcPartyId == null || patientForeignKeys == null || patientForeignKeys.getIds().size()==0) {
```
Comment (lespons) [inline src/main/java/org/taktik/icure/services/external/rest/v1/facade/ContactFacade.java]: `patientForeignKeys.getIds().size()==0`
we dont have file formatting watchers on backend ? :)

**Comment by alina-meyvn** [comment]
Add filtering on health elements by `tags` and/or by `status`

File: src/main/java/org/taktik/icure/services/external/rest/v1/dto/filter/healthelement/HealthElementByHcPartyTagCodeFilter.java [hunk 4]
```diff
@@ -1,72 +1,106 @@
...
+		this.tagCode = tagCode;
+	}
+
+	public void setStatus(int status) {
+		this.status = status;
+	}
+
+	@Override
+	public int hashCode() {
+		return Objects.hashCode(healthCarePartyId, codeType, codeNumber);
```
Comment (TheSleepyMonkey) [inline src/main/java/org/taktik/icure/services/external/rest/v1/dto/filter/healthelement/HealthElementByHcPartyTagCodeFilter.java]: missing the tags/status?

File: src/main/java/org/taktik/icure/services/external/rest/v1/dto/filter/healthelement/HealthElementByHcPartyTagCodeFilter.java [hunk 5]
```diff
@@ -1,72 +1,106 @@
...
+	public boolean equals(Object obj) {
+		if (this == obj) {
+			return true;
+		}
+		if (obj == null || getClass() != obj.getClass()) {
+			return false;
+		}
+		final HealthElementByHcPartyTagCodeFilter other = (HealthElementByHcPartyTagCodeFilter) obj;
+		return Objects.equal(this.healthCarePartyId, other.healthCarePartyId) && Objects.equal(this.codeType, other.codeType) && Objects.equal(this.codeNumber, other.codeNumber)
+				&& Objects.equal(this.tagType, other.tagType) && Objects.equal(this.tagCode, other.tagCode);
```
Comment (TheSleepyMonkey) [inline src/main/java/org/taktik/icure/services/external/rest/v1/dto/filter/healthelement/HealthElementByHcPartyTagCodeFilter.java]: Missing the status

File: src/main/java/org/taktik/icure/services/external/rest/v1/dto/filter/healthelement/HealthElementByHcPartyTagCodeFilter.java (hunk 4 above)
Comment (alina-meyvn) [reply src/main/java/org/taktik/icure/services/external/rest/v1/dto/filter/healthelement/HealthElementByHcPartyTagCodeFilter.java]: Yup. Thanks

**Pull request merged**